packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -124,22 +153,175 @@ const AccountRecoveryResetPassword = ({
+        supportsRecovery = false;
+      }
+    } catch (err) {
+      if (err instanceof ContextValidationErrors) {
+        contextError = err;
+        linkStatus = 'damaged';
+      }
+    }
+
+    return {
+      email,
+      linkStatus,
+      forceAuth,
+      supportsRecovery,
+      contextError,
+    };
+  }
+
+  async function onSubmit(data: FormData) {
+    const password = data.newPassword;
+
+    try {
+      await account.resetPasswordWithRecoveryKey(
+        {
+          password,
+          ...verificationInfo,
+          ...accountRecoveryKeyInfo
+        }
+      );
+
+      // FOLLOW-UP: I don't see functionality in settings
+      await account.setLastLogin(Date.now());
+
+      // FOLLOW-UP: It seems like the account class will now refresh itself, so I don't think
+      //            these are necessary anymore?
+      // storageContext.set('currentAccountUid', account.uid);
+      // await account.refresh('account');
+
+      // FOLLOW-UP: No equivalent yet in settings
+      notifier.onAccountSignIn(account);
+
+      relier.resetPasswordConfirm = true;
+      logViewEvent(viewName, 'verification.success');
+
+      // FOLLOW-UP: Broker not yet implemented
+      await broker.invokeBrokerMethod('afterCompleteResetPassword', account);
+      alertSuccess();
+      navigateAway();
+    } catch (err) {
+      if (AuthUiErrors['INVALID_TOKEN'].errno === err.errno) {
+        // TODO: Is this needed? Had to add the a method to support this.
+        // BEFORE: this.logError(err);
+        logErrorEvent({ viewName, ...err });
+        setLinkStatus('expired');
+      } else {
+        // Context validation errors indicate a bad state in either the url query or
+        // maybe storage. In these cases show an alert bar and let the error keep bubbling
+        // up.
+        if (err instanceof ContextValidationErrors) {
+          alertInvalidContext(err);
+        } else {
+          logErrorEvent(err);
+          const msg = ftlMsgResolver.getMsg(
+            'unexpected-error-encountered',
+            'Unexpected Error Encountered'
+          );
+          alertBar.error(msg);
+        }
+        throw err;
+      }
+    }
+  }
+
+  async function resendLinkHandler() {
+    logViewEvent(viewName, 'account-recovery-reset-password.resend');
+
+    // FOLLOW-UP: The previous code would retry the resend several times. Is that really necessary...
+    try {
+      await account.resetPassword(state.email);
+      const msg = ftlMsgResolver.getMsg(
+        `email-resent`,
+        'Email resent. Add accounts@firefox.com to your contacts to ensure a smooth delivery.'
+      );
+      alertBar.success(msg);
+    } catch (err) {
+      // FOLLOW-UP: Not sure if this is the right way to handle this. It didn't seem like the previous
+      //       code would not translate the error?
+      const msg = ftlMsgResolver.getMsg(
+        'email-resend-error',
+        'Sorry, there an issues resending!'
```

Comment:
Definitely a good idea to add localization for the error message, even if it wasn't done previously!

Nit: There is a typo in the message. :)

To follow the format used for other similar error messages in fxa-settings, maybe this should be:
"Sorry, there was a problem resending a reset password link to your email."